### DISCUSSION TITLE
Signalling a stop because NaNs detected in the Solution

### DISCUSSION THREAD

**Discussion by hzw-zju-2023** [AMReX-Combustion/PeleC#739] [Q&A]
I am trying to simulate liquid ammonia spray, but often encounter the error of Signalling a stop because NaNs detected in the Solution. I made the modification on the basis of an example of n-dodecane, and I found that when I refined the mesh, or adjusted some spray-related Settings, it was always easy to have this problem. Do not know how to find the origin of the problem in pelec, or how to avoid similar problems?

**Comment by marchdf** [comment]
@ldowen do you have any thoughts to help debug this?

**Comment by ldowen** [comment]
While it could be any number of things, my initial guess is that the liquid volume fraction is too high. I had intended to add a check to the injection routine to ensure the number density of the injected particles are set appropriately to avoid this issue. But it is something that should really be added. Basically, the number density of the injected droplets should be set to ensure the average injected droplet volume does not exceed the volume of the finest cell. Ideally, the droplet volume should not exceed 10% of the finest cell volume but that might result in very small number densities. Is it possible this is the issue you are experiencing?

**Comment by baperry2** [comment]
@hzw-zju-2023 this particular error message comes from a broad final check to make sure nothing has gone wrong at the end of each timestep, so it does not directly indicate what the problem is. However, if you are outputting plot files, there should be a plot file from the final timestep. If you visualize that and look at which variables have gone NaN and where in the domain, it may provide clues as to whether the source of the error is what @ldowen is describing or something else.
